utils/orchestrators.ts

```diff
@@ -101,78 +99,85 @@ export const checkUpdateCardIsRunning = (
     fiscalCode: FiscalCode,
     cardStatus: string
   ) => string = makeUpdateCgnOrchestratorId
-): TaskEither<CheckUpdateCardIsRunningErrorTypes, false> =>
-  isOrchestratorRunning(client, getOrchestratorId(fiscalCode, card.status))
-    .foldTaskEither<CheckUpdateCardIsRunningErrorTypes, false>(
-      err =>
-        fromLeft(
-          ResponseErrorInternal(
-            `Error checking UpdateCardOrchestrator: ${err.message}`
-          )
-        ),
-      ({ isRunning }) =>
-        isRunning ? fromLeft(ResponseSuccessAccepted()) : taskEither.of(false)
-    )
-    .chain(_ =>
-      taskEither.of(cgnStatuses.filter(el => el !== card.status.toString()))
-    )
-    .chain(otherStatuses =>
+): TE.TaskEither<CheckUpdateCardIsRunningErrorTypes, false> =>
+  pipe(
+    getOrchestratorId(fiscalCode, card.status),
+    orchestratorId => isOrchestratorRunning(client, orchestratorId),
+    TE.mapLeft(err =>
+      ResponseErrorInternal(
+        `Error checking UpdateCardOrchestrator: ${err.message}`
+      )
+    ),
+    TE.chainW(({ isRunning }) =>
+      isRunning ? TE.left(ResponseSuccessAccepted("", undefined)) : TE.of(false)
+    ),
+    TE.chainW(() =>
+      TE.of(cgnStatuses.filter(el => el !== card.status.toString()))
+    ),
+    TE.chainW(otherStatuses =>
       // check over other possible CGN' s statuses if there is other concurrent
       // orchestrators running. This check allows only one update's orchestrator
       // is running at once for each fiscalCode
-      array.sequence(taskEither)(
+      array.sequence(TE.ApplicativePar)(
         otherStatuses.map(status =>
-          isOrchestratorRunning(
-            client,
-            getOrchestratorId(fiscalCode, status)
-          ).foldTaskEither<CheckUpdateCardIsRunningErrorTypes, false>(
-            err =>
-              fromLeft(
-                ResponseErrorInternal(
-                  `Error checking UpdateCgnOrchestrator: ${err.message}`
-                )
-              ),
-            ({ isRunning }) =>
+          pipe(
+            getOrchestratorId(fiscalCode, status),
+            orchId => isOrchestratorRunning(client, orchId),
+            TE.mapLeft(err =>
+              ResponseErrorInternal(
+                `Error checking UpdateCgnOrchestrator: ${err.message}`
+              )
+            ),
+            TE.chainW(({ isRunning }) =>
               isRunning
-                ? fromLeft(
+                ? TE.left(
                     ResponseErrorConflict(
                       `Another Update Cgn orchestrator is running for status ${status}`
                     )
                   )
-                : taskEither.of(false)
+                : TE.of(false)
+            )
           )
         )
       )
-    )
-    .map(_ => false);
+    ),
+    TE.map(() => false as const)
```

Comment:
is `as const` needed? I tried and it compiles without it

Reply:
I've seen it also in other places...